plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -0,0 +1,490 @@
+	copy(originalQuotas, quotas)
+
+	atLeastOneChanged := false
+	for i := range admissionAttributes {
+		admissionAttribute := admissionAttributes[i]
+		newQuotas, err := e.checkRequest(quotas, admissionAttribute.attributes)
+		if err != nil {
+			admissionAttribute.result = err
+			continue
+		}
+
+		// if the new quotas are the same as the old quotas, then this particular one doesn't issue any updates
+		// that means that no quota docs applied, so it can get a pass
+		atLeastOneChangeForThisWaiter := false
+		for j := range newQuotas {
+			if !reflect.DeepEqual(originalQuotas[j], newQuotas[j]) {
+				atLeastOneChanged = true
+				atLeastOneChangeForThisWaiter = true
+				break
+			}
+		}
+
+		if !atLeastOneChangeForThisWaiter {
+			admissionAttribute.result = nil
+		}
+		quotas = newQuotas
+	}
+
+	// if none of the requests changed anything, there's no reason to issue an update, just fail them all now
+	if !atLeastOneChanged {
+		return
+	}
+
+	// now go through and try to issue updates.  Things get a little weird here:
+	// 1. check to see if the quota changed.  If not, skip.
+	// 2. if the quota changed and the update passes, be happy
+	// 3. if the quota changed and the update fails, add the original to a retry list
+	var updatedFailedQuotas []api.ResourceQuota
+	var lastErr error
+	nonConflictError := false
+	for i := range quotas {
+		newQuota := quotas[i]
+		// if this quota didn't change, skip it
+		if reflect.DeepEqual(originalQuotas[i], newQuota) {
+			continue
+		}
+
+		_, lastErr = e.client.Core().ResourceQuotas(newQuota.Namespace).UpdateStatus(&newQuota)
+		if lastErr != nil {
+			if !errors.IsConflict(lastErr) {
+				nonConflictError = true
+			}
+			updatedFailedQuotas = append(updatedFailedQuotas, newQuota)
+		}
+	}
+
+	if len(updatedFailedQuotas) == 0 {
+		// all the updates succeeded.  At this point, anything with the default deny error was just waiting to
+		// get a successful update, so we can mark and notify
+		for _, admissionAttribute := range admissionAttributes {
+			if IsDefaultDeny(admissionAttribute.result) {
+				admissionAttribute.result = nil
+			}
+		}
+		return
+	}
+
+	// this retry logic has the same bug that its possible to be checking against quota in a state that never actually exists where
+	// you've added a new documented, then updated and old one, your resource matches both and you're only checking one
+	if remainingRetries > 0 && !nonConflictError {
+		// updates for these quota names failed.  Get the current quotas in the namespace, compare by name, check to see if the
+		// resource versions have changed.  If not, we're going to fall through an fail everything.  If they all have, then we can try again
+		newQuotas, err := e.getQuotas(quotas[0].Namespace)
+		if err == nil {
+			quotasToCheck := []api.ResourceQuota{}
+			for _, newQuota := range newQuotas {
+				for _, oldQuota := range updatedFailedQuotas {
+					if newQuota.Name == oldQuota.Name {
```

Comment:
> in theory a new quota could have been added and we should also add the quotas that were not seen before, but i am not too worried about that scenario....

They could have been added, but I'm less sure that we should rescan unless we also run compensation logic to end up with proper point in time execution.  I'd rather say that quota prevents abuse, but doesn't give an always exact number

That's also consistent with existing edge cases.